src/eleveldb.erl

```diff
@@ -335,7 +338,13 @@ validate_options(Type, Opts) ->
                             validate_type(KType, V)
                     end, Opts).
 
+-spec property_cache(string(), string()) -> ok.
+property_cache(_BucketKey, _Properties) ->
+    erlang:nif_error({error, not_loaded}).
 
+-spec set_metadata_pid(pid()) -> ok.
+set_metadata_pid(_Pid) ->
+    erlang:nif_error({error, not_loaded}).
```

Comment:
I would prefer a generic interface here to register an external service pid so we don't need a new function each time we do this sort of thing, and I don't expect this to be the only service we're going to want to call from eleveldb (think lager).

For example:

```erlang
-spec register_service(Type :: atom(), Pid :: pid(), Info :: term()) -> {ok, reference()} | {error, term()}.
-spec unregister_service(Type :: atom(), Key :: pid() | reference()) -> ok | {error, term()}.
```

In this example,
```erlang
    set_metadata_pid(P)
```
might be called as
```erlang
    register_service(eleveldb_metadata, P, [])
```

I'm not sure about the `unregister` side, just throwing something out for the case where we detect some service is unavailable. It seems a ref could be associated with a registration that could subsequently be used to unregister that service.